File: backend/BYTUBE/Models/CommentModel.cs
using BYTUBE.Models.UserModels;

namespace BYTUBE.Models
{
    public class CommentModel
    {
        public string? Id { get; set; }

        public required string Message { get; set; }

        public string? UserId { get; set; }

        public string VideoId { get; set; }

        public DateTime? Created { get; set; }

        public int? LikesCount { get; set; }
        public bool? UserIsLikeIt { get; set; }
        public bool? IsVideoOwner { get; set; }

        public UserPublicModel? User {  get; set; }
    }
}

File: frontend/src/types/models/CommentModel.ts
import { User } from "./UserModel";

interface CommentModel {
  id?: string;
  message: string;
  userId: string;
  videoId: string;
  likesCount?: number;
  userIsLikeIt?: boolean;
  isVideoOwner?: boolean;
  created?: string;

  user?: User;
}

export default CommentModel;
